### PR TITLE
JSlint check

### PR DESCRIPTION
**Pull request by frogsandsocks**
Added jslint support to Travis CI build.

### CODE REVIEW

File: .jslintrc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    "gettext",
+    "windowname_to_id",
+    "jQuery",
+    "WebSocket",
+    "$",
+    "dismissRelatedLookupPopup",
+    "dismissAddAnotherPopup",
+    "FileReader",
+    "Image",
+    "L",
```
Comment (nemesisdesign) [inline .jslintrc]: remove the previous 5 lines

File: .travis.yml [hunk 2]
```diff
@@ -25,12 +28,24 @@ before_install:
   - ./runflake8
   - ./runisort
 
+  # Install NVM:
+  - curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.0/install.sh | bash
```
Comment (nemesisdesign) [inline .travis.yml]: we did a similar tasks in other modules and this was not needed, we want to keep the build step as simple and fast as possible, therefore I kindly ask to please remove it

File: .travis.yml [hunk 3]
```diff
@@ -25,12 +28,24 @@ before_install:
...
+
+  # Install nodeJS:
+  - nvm install v8
```
Comment (nemesisdesign) [inline .travis.yml]: same as previous comment

File: .travis.yml [hunk 4]
```diff
@@ -25,12 +28,24 @@ before_install:
...
+
 install:
   - pip install $DJANGO
   - python setup.py -q develop
 
+  # Install JSlint:
+  - npm install --save jslint -g
```
Comment (nemesisdesign) [inline .travis.yml]: move this in before_install section

File: .travis.yml [hunk 5]
```diff
@@ -25,12 +28,24 @@ before_install:
...
+
 script:
   - coverage run --source=django_netjsongraph runtests.py
 
+  # JSlint check:
+  - jslint --config ./.jslintrc ./django_netjsongraph/static/netjsongraph/js/*.js
```
Comment (nemesisdesign) [inline .travis.yml]: move this in before_install section, after installing jslint
we do this to stop the build if any style check fails before installing the other python dependencies, which speeds up a lot the style checks and failures are reported very early if they happen

File: .travis.yml [hunk 6]
```diff
@@ -6,6 +6,9 @@ python:
   - "3.4"
   - "2.7"
 
+node_js:
+  - "8"
```
Comment (nemesisdesign) [inline .travis.yml]: this should not be needed (we have added similar checks in other modules without adding this), try to remove it

**Comment by frogsandsocks** [comment]
@nemesisdesign 
Now commit looks like the example in contribution guidelines.

**Comment by frogsandsocks** [comment]
@nemesisdesign 
I fixed some jslint check errors, if you don't need this commit i can delete it

**Comment by frogsandsocks** [comment]
I fixed some jslint check errors, if you don't need this commit i can delete it

File: .travis.yml [hunk 7]
```diff
@@ -25,6 +25,12 @@ before_install:
   - ./runflake8
   - ./runisort
 
+  # Install JSlint:
+  - npm install --save jslint -g
+
```
Comment (nemesisdesign) [inline .travis.yml]: remove blank line, also remove the comments because they are not necessary (it's pretty obvious that we are installing jslint and we are running a jslint check), also remove the blank line after `- ./runisort`

File: .jslintrc [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  "validthis":  true,
+  "strict_mode":true,
+
+  "browser":    false,
+  "devel":      false,
+  "node":       false,
+  "rhino":      false,
+  "widget":     false,
+  "windows":    false,
+  "indent":     2,
```
Comment (nemesisdesign) [inline .jslintrc]: please use 4 spaces indentation and revert the changes to the JS files because that kind of huge change breaks `git blame` (because you appear author of those lines while you are not, so if something breaks somebody may think you are the person to ask why something is not working)

**Comment by frogsandsocks** [comment]
I can't claim this task again, because deadline is over. But i don't care about code-in, so you can just ignore it

**Comment by nemesisdesign** [comment]
@frogsandsocks do you want to finish this PR or should I close it?

**Comment by nemesisdesign** [comment]
@frogsandsocks sorry, didn't check that you did address my comments, thanks :+1:

**Comment by frogsandsocks** [comment]
@nemesisdesign 
Yes, i want to complete this task, but i don't care about code-in
